src/services/userToRoomJoiner.js

```diff
@@ -0,0 +1,61 @@
+const { User, Room } = require('../entities')
+
+class UserToRoomJoiner {
+    constructor({ userRepo, roomRepo }) {
+        this.userRepo = userRepo
+        this.roomRepo = roomRepo
+    }
+
+    call({ username, roomTitle }) {
```

Comment:
Why is only this method wrapper in try/catch and not the others?

Reply:
Appeared(Throwing) error goes up through the callstack until it is in try-catch block, than catch block catches the error
https://www.javascripttutorial.net/javascript-call-stack/

Reply:
So can we say that if some method is only used in another methods `try` we can omit the try/catch for it?

And if we're talking about the code in this file, will it work something like this:
`call` is waiting for whatever `join` will return to it (in `try`)
`join` is waiting for whatever `getUser` will return to it (to be eshort let's say that nothing more is called)
and then the `getUser` returns an error to `join`, and `join` passes it on to the `call`.
And the `call` method decides what to do with it?

Is this 'train of thought' correct?

So it won't actually be the app stopping at the `getUser`, instead it will actually "go up" a couple methods?

Reply:
The app is not stopped in case of promises/async-await/async-cb . It is handled by node.js event loop.  Not sure that will be clean enough, but lets try https://www.freecodecamp.org/news/nodejs-eventloop-tutorial . You may find something more clear